mockwebserver/src/main/java/okhttp3/mockwebserver/MockWebServer.kt

```diff
@@ -270,7 +270,7 @@ class MockWebServer : ExternalResource(), Closeable {
    * @return the head of the request queue
    */
   @Throws(InterruptedException::class)
-  fun takeRequest(timeout: Long, unit: TimeUnit): RecordedRequest? =
+  fun takeRequest(timeout: Long, unit: TimeUnit): RecordedRequest =
```

Comment:
Don't think we want to make this change. It changes the behavior from 3.x which returned null (despite saying it never did) to throwing NPE. I'm comfortable with this source incompatibility.

Reply:
Thanks, reverted.